crates/host/src/event.rs

```diff
@@ -1,15 +1,33 @@
 use std::collections::{BTreeMap, HashMap};
 
-use anyhow::Context;
-use cloudevents::{EventBuilder, EventBuilderV10};
 use serde_json::json;
-use time::{format_description::well_known::Rfc3339, OffsetDateTime};
-use tracing::{instrument, warn};
-use ulid::Ulid;
-use uuid::Uuid;
 use wascap::jwt;
 use wasmcloud_control_interface::Link;
 
+/// A trait for publishing wasmbus events. This can be implemented by any transport or bus
+/// implementation that can send the serialized event to the appropriate destination.
+///
+/// TODO(brooksmtownsend): file an issue for this: This trait can certainly be enhanced by adding methods specific to the event
+#[async_trait::async_trait]
+pub trait EventPublisher: Send + Sync {
+    /// Publish an event that occurred in the host. The event name is the type of event being published, and the
+    /// data is the payload of the event. It's up to the implementation to
+    /// determine how to handle events. By default, this is a no-op.
+    async fn publish_event(
```

Comment:
If this trait is ever going to be a `Box`ed `dyn`ner, than ignore this, but it could be good to just make `data` be  `T where T: serde::Serialize`